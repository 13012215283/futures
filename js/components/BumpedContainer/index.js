import React, { Component } from 'react';
import { View, Keyboard } from 'react-native';
import PropTypes from 'prop-types';

import styles from './style';

export default class BumpedContainer extends Component {
  static defaultProps = {
    initPos: 0,
    bumptPos: 0,
  };

  constructor(props) {
    super(props);

    this.state = {
      keyboardShow: false,
    };

    this.keyboardShowListener = null;
    this.keyboardHideListener = null;

    this.keyboardDidShowHandler = this.keyboardDidShowHandler.bind(this);
    this.keyboardDidHideHandler = this.keyboardDidHideHandler.bind(this);
  }

  componentWillMount() {
    this.keyboardShowListener = Keyboard.addListener(
      'keyboardDidShow',
      this.keyboardDidShowHandler
    );

    this.keyboardHideListener = Keyboard.addListener(
      'keyboardDidHide',
      this.keyboardDidHideHandler
    );
  }

  keyboardDidShowHandler() {
    this.setState({
      keyboardShow: true,
    });
  }

  keyboardDidHideHandler() {
    this.setState({
      keyboardShow: false,
    });
  }

  render() {
    const { initPos, bumptPos } = this.props;

    return (
      <View
        style={[
          styles.inputArea,
          { top: initPos },
          this.state.keyboardShow && {
            top: bumptPos,
          },
        ]}
      >
        {this.props.children}
      </View>
    );
  }
}

BumpedContainer.propTypes = {
  children: PropTypes.element.isRequired,
  initPos: PropTypes.number,
  bumptPos: PropTypes.number,
};
