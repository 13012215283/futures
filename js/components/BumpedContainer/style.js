import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  inputArea: {
    position: 'absolute',
    width: '100%',
    paddingLeft: 10,
    paddingRight: 10,
  },
});

export default styles;
