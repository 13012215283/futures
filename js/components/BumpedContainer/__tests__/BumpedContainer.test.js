import { View, Text } from 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import BumpedContainer from '../index';

/**
 * @property initPos 初始展现时距离顶部的距离
 * @property bumptPos 弹框弹出时距离顶部的距离
 */

test('BumpedContainer renders correctly', () => {
  const tree = renderer
    .create(
      <BumpedContainer initPos={45} bumptPos={10}>
        <View>
          <Text>test</Text>
        </View>
      </BumpedContainer>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('if set property novalue, it render correctly', () => {
  const tree = renderer
    .create(
      <BumpedContainer>
        <View>
          <Text>test</Text>
        </View>
      </BumpedContainer>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
