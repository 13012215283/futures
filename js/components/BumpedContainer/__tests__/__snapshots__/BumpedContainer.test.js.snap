// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BumpedContainer renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingLeft": 10,
        "paddingRight": 10,
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "top": 45,
      },
      false,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      test
    </Text>
  </View>
</View>
`;

exports[`if set property novalue, it render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingLeft": 10,
        "paddingRight": 10,
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "top": 0,
      },
      false,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      test
    </Text>
  </View>
</View>
`;
